rest/route/user_test.go

```diff
@@ -1011,3 +1011,89 @@ func TestOffboardUserHandlerAdminis(t *testing.T) {
 		assert.NotContains(t, projRef.Admins, offboardedUser)
 	}
 }
+
+func TestGetUserHandler(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	assert.NoError(t, db.ClearCollections(user.Collection))
+	usrToRetrieve := user.DBUser{
+		Id:           "beep.boop",
+		DispName:     "robots are good",
+		EmailAddress: "bots_r@us.com",
+		APIKey:       "secret_key",
+		PatchNumber:  12,
+		OnlyAPI:      true,
+		SystemRoles: []string{
+			"bot_access",
+		},
+	}
+	me := user.DBUser{
+		Id:           "me",
+		EmailAddress: "i@rock.com",
+		APIKey:       "my_key",
+	}
+	assert.NoError(t, usrToRetrieve.Insert())
+	assert.NoError(t, me.Insert())
+
+	for testName, testCase := range map[string]func(t *testing.T){
+		"UserNotFound": func(t *testing.T) {
+			req, err := http.NewRequest(http.MethodGet, "http://example.com/api/rest/v2/users/no_one", nil)
+			req = gimlet.SetURLVars(req, map[string]string{"user_id": "no_one"})
+
+			require.NoError(t, err)
+			handler := makeGetUserHandler()
+
+			assert.NoError(t, handler.Parse(ctx, req))
+			assert.Equal(t, handler.(*getUserHandler).userId, "no_one")
+
+			resp := handler.Run(gimlet.AttachUser(context.Background(), &me))
```

Comment:
Pass a context into the test rather than use `context.Background()` directly (even if it's just `context.WithCancel(context.Background())`).

Reply:
I have some stuff in the unofficial style guide but basically, it's most problematic if the test isn't fast or the implementation _could_ use a goroutine somewhere. We've seen this happen in other repos (e.g. Jasper) where a lot of tests depend on being "fast enough" or especially if it uses goroutines, where the test can just keep running during later tests if the context doesn't cancel because Go will time out the test but the background context will let it keep running until `go test` exits. This causes a lot of problems later on because other tests could be invisibly running alongside other tests after they've already failed.

I don't particularly know when we might introduce goroutines to things since we could do it at anytime, but adding a `context.WithCancel` I find simple enough to prevent the issue from coming up. Go also discourages using a plain `context.Background()` as much as possible for reasons like goroutines.